ocrd_anybaseocr/cli/ocrd_anybaseocr_textline.py

```diff
@@ -78,18 +84,38 @@ def process(self):
             page = pcgts.get_Page()
             LOG.info("INPUT FILE %s", input_file.pageId or input_file.ID)
             
-            page_image, page_xywh, page_image_info = self.workspace.image_from_page(page, page_id, feature_filter='tiseged')            
+            page_image, page_xywh, page_image_info = self.workspace.image_from_page(page, page_id, feature_filter='tiseged')
+            
             if oplevel == 'page':
-                self._process_segment(page_image, page, page_xywh, page_id, input_file, n)
-                LOG.warning("Operation level should not be page.")
+                LOG.warning("Operation level should be region.")
+                break
             else:
-                regions = page.get_TextRegion() + page.get_TableRegion()
+                regions = page.get_TextRegion()
                 if not regions:
                     LOG.warning("Page '%s' contains no text regions", page_id)
+                    continue
                 for (k, region) in enumerate(regions):
-                    region_image, region_xywh = self.workspace.image_from_segment(region, page_image, page_xywh)            
+                    
+                    points = region.Coords.get_points()
+                    points = points.split(" ")
+                    
+                    x_min = min(int(points[0].split(",")[0]), int(points[1].split(",")[0]), int(points[2].split(",")[0]), int(points[3].split(",")[0]))
+                    x_max = max(int(points[0].split(",")[0]), int(points[1].split(",")[0]), int(points[2].split(",")[0]), int(points[3].split(",")[0]))
+                    y_min = min(int(points[0].split(",")[1]), int(points[1].split(",")[1]), int(points[2].split(",")[1]), int(points[3].split(",")[1]))
+                    y_max = max(int(points[0].split(",")[1]), int(points[1].split(",")[1]), int(points[2].split(",")[1]), int(points[3].split(",")[1]))
+
+                    if x_max>page_image.size[0]:
+                        x_max = page_image.size[0]-1
+                    if y_max>page_image.size[1]:
+                        y_max = page_image.size[1]-1
+                    
+                    img__ = page_image.crop((x_min,y_min,x_max,y_max))
```

Comment:
the problem was exactly that it already had an alternative image
we tried to "merge" the following two pipelines:
`raw image --> binarization --> deskew --> cropping --> tiseg  ---> textline` with
`raw image --> block segmentation --(region+alternative image)--> textline               ` 
per default block segmentation adds an alternative image for each extracted region which means textline will take the alternative image from the raw image over applying the extracted regions to the binarized/deskewed etc. image (both of which textline needs)
We are aware that the pipeline was not supposed to have multiple branches, but with the dependencies of our methods there is no way for a single straight forward pipeline, so we tried to use the API as best as possible.
we ended up with three choices: 
either uncouple the pipelines and perform: `raw image --> block segmentation --> binarization --> deskew --> textline` independently, apply the regions manually (which we tried to do) or not add an alternative image during block segmentation (which may be better?)
What do you think would be the best option? (We tried to explain the problem before, and I hope its understandable now, but communicating the problem turned out to be difficult.)

Reply:
> per default block segmentation adds an alternative image for each extracted region which means textline will take the alternative image from the raw image over applying the extracted regions to the binarized/deskewed etc. image (both of which textline needs)

This is not a valid sentence in English, and I don't understand it.

> We are aware that the pipeline was not supposed to have multiple branches, but with the dependencies of our methods there is no way for a single straight forward pipeline, so we tried to use the API as best as possible.

What the API can give you are filters and selector on operations already performed earlier in the workflow. It cannot provide the workflow itself for a processor! (That's up to the user to decide. But of course, you can _document_ what a good setup is for your tools.) 

However, the processor can constrain its input images, so it does not pick up the wrong annotations when there is a freedom of choice, and stops execution when put in the wrong (place of a) pipeline.

So I don't see a problem with the above workflow. Just that your textline segmentation needs 2 input file groups then (to merge their regions I guess). 

You could even be _very_ restrictive to prevent other workflows:
- binarization: filter `binarized,deskewed,cropped` page images
- deskewing: select `binarized`, filter `deskewed,cropped` page images
- cropping: select `binarized,deskewed`, filter `cropped` page images
- tiseg: select `binarized,deskewed,cropped` page images
- block segmentation: filter `binarized,deskewed,cropped` page images
- textline: select `binarized,deskewed,cropped` region images on input file group 1, filter  `binarized,deskewed,cropped` region images on input file group 2

But you don't _need_ to do this (and I would recommend against it, except where an algorithm just cannot handle the input otherwise).

> We tried to explain the problem before, and I hope its understandable now, but communicating the problem turned out to be difficult.

You can always turn to the Gitter Lobby for general questions.

Reply:
> The `region_xywh` you are passing to `_process_segment` along with `img__` does not match. It **will** give wrong coordinates.

> What made you bypass `image_from_segment`'s `region_image`?

This has been resolved by 13d855efd5d51670922c70ff7358a84e27243b3e (although the wrong method still remains as commented code).

But the rest of the discussion was about filters/selectors. You took my advice to the extreme in 0ac8567b1faffdd8bc5e1c67a74fb03761404661 – hence my comment [below](https://github.com/mjenckel/ocrd_anybaseocr/pull/27#pullrequestreview-333207653)